posthog/async_migrations/migrations/0006_persons_and_groups_on_events_backfill.py

```diff
@@ -0,0 +1,259 @@
+                """,
+                rollback=f"DROP TABLE IF EXISTS {TEMPORARY_PERSONS_TABLE_NAME}",
+                per_shard=True,
+            ),
+            AsyncMigrationOperationSQL(
+                sql=f"""
+                    CREATE TABLE {TEMPORARY_PDI2_TABLE_NAME}
+                    (
+                        `team_id` Int64,
+                        `distinct_id` String,
+                        `person_id` UUID,
+                        `is_deleted` Int8,
+                        `version` Int64 DEFAULT 1
+                    )
+                    ENGINE = ReplacingMergeTree(version)
+                    ORDER BY (team_id, distinct_id)
+                    SETTINGS index_granularity = 128
+                """,
+                rollback=f"DROP TABLE IF EXISTS {TEMPORARY_PDI2_TABLE_NAME}",
+                per_shard=True,
+            ),
+            AsyncMigrationOperationSQL(
+                sql=f"""
+                    CREATE TABLE {TEMPORARY_GROUPS_TABLE_NAME}
+                    (
+                        `group_type_index` UInt8,
+                        `group_key` String,
+                        `created_at` DateTime64(3),
+                        `team_id` Int64,
+                        `group_properties` String,
+                        `_timestamp` DateTime
+                    )
+                    ENGINE = ReplacingMergeTree(_timestamp)
+                    ORDER BY (team_id, group_type_index, group_key)
+                    SETTINGS index_granularity = 128
+                """,
+                rollback=f"DROP TABLE IF EXISTS {TEMPORARY_GROUPS_TABLE_NAME}",
+                per_shard=True,
+            ),
+            AsyncMigrationOperationSQL(
+                sql=f"""
+                    INSERT INTO {TEMPORARY_PERSONS_TABLE_NAME} (id, created_at, team_id, properties, is_deleted, version)
+                    SELECT id, created_at, team_id, properties, is_deleted, version FROM person
+                """,
+                rollback=None,
+                per_shard=True,
+            ),
+            AsyncMigrationOperationSQL(
+                sql=f"""
+                    INSERT INTO {TEMPORARY_PDI2_TABLE_NAME} (team_id, distinct_id, person_id, is_deleted, version)
+                    SELECT team_id, distinct_id, person_id, is_deleted, version FROM person_distinct_id2
+                """,
+                rollback=None,
+                per_shard=True,
+            ),
+            AsyncMigrationOperationSQL(
+                sql=f"""
+                    INSERT INTO {TEMPORARY_GROUPS_TABLE_NAME} (group_type_index, group_key, created_at, team_id, group_properties)
+                    SELECT group_type_index, group_key, created_at, team_id, group_properties FROM groups
+                """,
+                rollback=None,
+                per_shard=True,
+            ),
+            AsyncMigrationOperation(
+                fn=lambda query_id: run_optimize_table(
```

Comment:
Not a fan of this function: I now need to reverse-engineer the SQL this generates to make use of it.

I'd much rather operate with raw SQL everywhere instead of building a pseudo-orm.